pandas/core/series.py

```diff
@@ -232,7 +232,10 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
 
         generic.NDFrame.__init__(self, data, fastpath=True)
 
-        object.__setattr__(self, 'name', name)
+        if name is not None and not com.is_hashable(name):
+            raise TypeError('Series.name must be a hashable type')
+        object.__setattr__(self, '_name', name)
+
```

Comment:
you are duplicating code here.

just ``self.name = name`` (this will then hit the setattr property)